src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -14521,8 +14527,16 @@ void Unit::SetHealth(uint32 val)
             val = maxHealth;
     }
 
+    float prevHealthPct = GetHealthPct();
+
     SetUInt32Value(UNIT_FIELD_HEALTH, val);
 
+    // mobs that are now or were below 30% need to update their speed
+    if (GetTypeId() == TYPEID_UNIT && (prevHealthPct < 30.0 || HealthBelowPct(30)))
```

Comment:
You need here two checks:
1. if mob were below 30 and now is above it
2. if mob were above 30 and now is below it

Reply:
I don't see why? The formula is continuous but breaks at 30. So unless we want to make it by steps, eg 30-25-20-15-10, I think this is correct?
If a mob was previously at 30, if he got lower than before I need to recompute. If he got higher, I need to recompute as well (my understanding is they gain speed back). Therefore `prevHealthPct < 30.0` 
If a mob is now below 30, he needs to be recomputed. So `HealthBelowPct(30)`.

And as it's both of these cases that need to re-evaluate, we use `||`. 

Did I misunderstand what we try to achieve?

Reply:
Okay - you are right.